tt_metal/impl/program/program.hpp

```diff
@@ -212,6 +216,9 @@ class Program {
 
     void add_semaphore(const CoreRangeSet & crs, uint32_t address, uint32_t init_value, CoreType core_type=CoreType::WORKER);
 
+    friend void detail::AddConfigTensor(Program &program, const Tensor& config_tensor);
+    void add_config_tensor(const Tensor& config_tensor);
+
```

Comment:
Isn't tensor an eager object and we want eager/metal to be separate?
Should we track buffers here instead?

Reply:
Yeah needed access to `Tensor`. You mean instead add the buffer pointer?

Reply:
Let me try with buffer ptr instead.

Reply:
Previously we had a model level cache for configs, which we got rid of since we want it to be hidden from model user, so we had decided to add it to the program itself.

Reply:
is there some intermediate level where this info can be kept with the op?

Reply:
This seems like it should belong with / leverage the program cache infra? Since you are associating a program with specific tensor buffers